common/build-style/void-cross.sh

```diff
@@ -0,0 +1,539 @@
+#
+# This helper is for void system crosstoolchain templates.
+#
+# Mandatory variables:
+#
+# - cross_triplet - the target triplet (e.g. aarch64-linux-gnu)
+# - cross_wordsize - 32 or 64
+# - cross_libc - glibc or musl
+# - cross_binutils_version
+# - cross_gcc_version
+# - cross_linux_version
+# - cross_linux_arch - the source ARCH of the kernel (e.g. arm64)
+# - cross_glibc_version
+# - cross_musl_version
+# - cross_libucontext_version - only on musl without cross_gcc_skip_go
+# - cross_libucontext_arch - ditto, the libucontext arch
+#
+# Optional variables:
+#
+# - cross_gcc_skip_go - do not build gccgo support
+# - cross_binutils_configure_args
+# - cross_gcc_bootstrap_configure_args
+# - cross_gcc_configure_args
+# - cross_glibc_cflags
+# - cross_glibc_ldflags
+# - cross_glibc_configure_args
+# - cross_musl_cflags
+# - cross_musl_ldflags
+# - cross_musl_configure_args
+#
+# configure_args is passed to both bootstrap gcc and final gcc
+# if you need to pass some to one and not the other, use the
+# respective cross_ variables for final gcc and bootstrap gcc
+#
+
+_apply_patch() {
+	local args="$1" pname="$(basename $2)"
+	if [ ! -f ".${pname}_done" ]; then
+		patch -N $args -i $2
+		touch .${pname}_done
+	fi
+}
+
+_check_cross_dir() {
+	local dname=$1
+	local sname=$2
+	local var="cross_${dname}_version"
+	cd ${wrksrc}
+	if [ -f ".${sname:=dname}_done" ]; then
+		return 0
+	fi
+	if [ ! -d "${1}-${!var}" ]; then
+		msg_error "${1}-${!var} does not exist"
+	fi
+}
+
+_build_binutils() {
+	_check_cross_dir binutils
+
+	msg_normal "Building binutils for ${cross_triplet}\n"
+
+	mkdir -p binutils_build
+	cd binutils_build
+
+	../binutils-${cross_binutils_version}/configure \
+		--prefix=/usr \
+		--sbindir=/usr/bin \
+		--libdir=/usr/lib \
+		--libexecdir=/usr/lib \
+		--target=${cross_triplet} \
+		--with-sysroot=/usr/${cross_triplet} \
+		--disable-nls \
+		--disable-shared \
+		--disable-multilib \
+		--disable-werror \
+		--disable-gold \
+		--with-system-zlib \
```

Comment:
If you know how, maybe force it to error out when it finds textrels for musl binaries? I didn't manage to make it work when I tried.